lib/tooling/readers/elf-parser.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from 'fs';
+
+import {assert} from '../../assert';
+
+import {DwarfLineReader, LineInfoItem} from './dwarf-line-reader';
+import {ElfReader} from './elf-reader';
+
+export {LineInfoItem} from './dwarf-line-reader';
+
+function pad(num_str: string) {
+    const s = '00000000' + num_str;
+    return s.substring(num_str.length);
+}
+
+export class ElfParser {
+    protected declare elfPath: string;
```

Comment:
It would not be initialize in constructor. Since `elfPath` is a dynamic data field of the `ElfParser`, i.e. it will changed by some functions(like `bindFile`), we should not treat it a attribute of `ElfParser`.